modules/apps/users-admin/users-admin-web/src/main/java/com/liferay/users/admin/web/internal/display/context/ViewTreeManagementToolbarDisplayContext.java

```diff
@@ -506,15 +506,30 @@ else if (className.equals(User.class.getName())) {
 					navigationStatus, null));
 		}
 		else {
-			searchContainer.setResultsAndTotal(
-				() -> OrganizationLocalServiceUtil.getOrganizationsAndUsers(
+			List<Object> organizationsAndUsers = new ArrayList<>();
+
+			organizationsAndUsers.addAll(
+				OrganizationLocalServiceUtil.getOrganizations(
 					_themeDisplay.getCompanyId(),
-					_organization.getOrganizationId(), navigationStatus,
-					searchContainer.getStart(), searchContainer.getEnd(),
-					searchContainer.getOrderByComparator()),
+					_organization.getOrganizationId()));
+
+			organizationsAndUsers.addAll(
```

Comment:
What about the navigationStatus filter? You are only using it in the counting but not in the search itself